generate.ts

```diff
@@ -306,14 +293,14 @@ function handleClass(cls: FuncContainer): string {
     }
     let name = cls.name;
     let inherits = "";
-    if (ENTITY_CHILDREN.includes(cls.name)) {
-        inherits = " : " + getTypeName("Entity");
+    if (PARENT_OVERRIDES[cls.name]) {
+        inherits = " : " + getTypeName(PARENT_OVERRIDES[cls.name]);
     } else if (cls.parent){
         inherits = " : " + getTypeName(cls.parent);
     }
     let def = `--- @class ${name}${inherits}\n`;
     let lua = `local ${name} = {}\n`;
-    return def + desc + lua;
+    return desc + def + lua;
```

Comment:
it wasnt howing docs properly with the description after the class definition so i put it before